graylog2-server/src/test/java/org/graylog2/indexer/EventsIndexMappingTest.java

```diff
@@ -47,59 +49,61 @@ void createsValidMappingTemplates(final String versionString) throws Exception {
-        at.jsonPathAsString("$.mappings.properties.timerange_end.format").isEqualTo(DATE_FORMAT);
-        at.jsonPathAsString("$.mappings.properties.streams.type").isEqualTo("keyword");
-        at.jsonPathAsString("$.mappings.properties.source_streams.type").isEqualTo("keyword");
-        at.jsonPathAsString("$.mappings.properties.message.type").isEqualTo("text");
-        at.jsonPathAsString("$.mappings.properties.message.analyzer").isEqualTo("standard");
-        at.jsonPathAsBoolean("$.mappings.properties.message.norms").isFalse();
-        at.jsonPathAsString("$.mappings.properties.message.fields.keyword.type").isEqualTo("keyword");
-        at.jsonPathAsString("$.mappings.properties.source.type").isEqualTo("keyword");
-        at.jsonPathAsString("$.mappings.properties.key.type").isEqualTo("keyword");
-        at.jsonPathAsString("$.mappings.properties.key_tuple.type").isEqualTo("keyword");
-        at.jsonPathAsString("$.mappings.properties.priority.type").isEqualTo("long");
-        at.jsonPathAsString("$.mappings.properties.alert.type").isEqualTo("boolean");
-        at.jsonPathAsString("$.mappings.properties.fields.type").isEqualTo("object");
-        at.jsonPathAsBoolean("$.mappings.properties.fields.dynamic").isTrue();
-        at.jsonPathAsString("$.mappings.properties.triggered_jobs.type").isEqualTo("keyword");
+        at.jsonPathAsString("$.properties.id.type").isEqualTo("keyword");
+        at.jsonPathAsString("$.properties.event_definition_type.type").isEqualTo("keyword");
+        at.jsonPathAsString("$.properties.event_definition_id.type").isEqualTo("keyword");
+        at.jsonPathAsString("$.properties.origin_context.type").isEqualTo("keyword");
+        at.jsonPathAsString("$.properties.timestamp.type").isEqualTo("date");
+        at.jsonPathAsString("$.properties.timestamp.format").isEqualTo(DATE_FORMAT);
+        at.jsonPathAsString("$.properties.timestamp_processing.type").isEqualTo("date");
+        at.jsonPathAsString("$.properties.timestamp_processing.format").isEqualTo(DATE_FORMAT);
+        at.jsonPathAsString("$.properties.timerange_start.type").isEqualTo("date");
+        at.jsonPathAsString("$.properties.timerange_start.format").isEqualTo(DATE_FORMAT);
+        at.jsonPathAsString("$.properties.timerange_end.type").isEqualTo("date");
+        at.jsonPathAsString("$.properties.timerange_end.format").isEqualTo(DATE_FORMAT);
+        at.jsonPathAsString("$.properties.streams.type").isEqualTo("keyword");
+        at.jsonPathAsString("$.properties.source_streams.type").isEqualTo("keyword");
+        at.jsonPathAsString("$.properties.message.type").isEqualTo("text");
+        at.jsonPathAsString("$.properties.message.analyzer").isEqualTo("standard");
+        at.jsonPathAsBoolean("$.properties.message.norms").isFalse();
+        at.jsonPathAsString("$.properties.message.fields.keyword.type").isEqualTo("keyword");
+        at.jsonPathAsString("$.properties.source.type").isEqualTo("keyword");
```

Comment:
I know it was like that before... and may be a task for different PR, in the cooldown phase... But wouldn't it be nicer to somehow construct the whole expected object and use it in assertion, instead of working with so many elements one by one?

Reply:
I am not sure. Working with the exact, complete object is always prone to make the test brittle when unrelated parts are changed.